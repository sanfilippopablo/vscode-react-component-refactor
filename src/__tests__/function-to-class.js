const pluginTester = require("babel-plugin-tester");
const functionToClass = require("../function-to-class");

pluginTester({
  plugin: functionToClass,
  tests: [
    {
      code: `
        const MyComponent = ({ a, b, c }) => {
          const result = a + b + c;
          return <div>{result}</div>;
        };
      `,
      snapshot: true
    },
    {
      code: `
        const MyComponent = ({ a, b, c }) => <div>hello</div>;
      `,
      snapshot: true
    },
    {
      code: `
        const MyComponent = () => <div>hello</div>;
      `,
      // snapshot: true
      output: `
        class 
      `
    }
  ],
  babelOptions: { babelrc: true, filename: ".babelrc" }
});
