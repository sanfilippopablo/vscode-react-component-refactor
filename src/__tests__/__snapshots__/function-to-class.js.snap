// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function-to-class 1. function-to-class: 1. function-to-class 1`] = `
"
const MyComponent = ({ a, b, c }) => {
  const result = a + b + c;
  return <div>{result}</div>;
};

      ↓ ↓ ↓ ↓ ↓ ↓

class MyComponent extends React.Component {
  render() {
    const { a, b, c } = this.props;

    const result = a + b + c;
    return <div>{result}</div>;
  }

}
"
`;

exports[`function-to-class 2. function-to-class: 2. function-to-class 1`] = `
"
const MyComponent = ({ a, b, c }) => <div>hello</div>;

      ↓ ↓ ↓ ↓ ↓ ↓

class MyComponent extends React.Component {
  render() {
    const { a, b, c } = this.props;
    return <div>hello</div>;
  }

}
"
`;
